sanic/response/convenience.py

```diff
@@ -331,3 +346,94 @@ async def _streaming_fn(response):
         headers=headers,
         content_type=mime_type,
     )
+
+
+class AutoIndex:
+    STYLE = """
+        html { font-family: sans-serif; }
+        main { padding: 1rem; }
+        table { width: 100%; max-width: 1200px; }
+        td { font-family: monospace; }
+        table.autoindex td:first-child { width: 65% }
+        table.autoindex td:last-child { text-align: right; width: 15%; }
+        span.icon { margin-right: 1rem; }
+        @media (prefers-color-scheme: dark) {
+            html { background: #111; color: #ccc; }
+            a { color: #ccc; }
+            a:visited { color: #777; }
+        }
+    """
+    TITLE = "📁 File browser"
+
+    def __init__(
+        self, directory: Path, autoindex: bool, index_name: str
+    ) -> None:
+        self.directory = directory
+        self.autoindex = autoindex
+        self.index_name = index_name
+
+    async def handle(self):
+        index_file = self.directory / self.index_name
+        if self.autoindex and (not index_file.exists() or not self.index_name):
+            return await self.index()
+
+        if self.index_name:
+            return await file(index_file)
+
+    async def index(self):
+        return html(self.render())
+
+    def render(self) -> str:
+        doc = Document(title=self.TITLE, lang="en")
+        self._head(doc)
+        with doc.main:
+            self._headline(doc)
+            self._file_table(doc)
+        return str(doc)
+
+    def _head(self, doc: Document):
+        doc.head.title(self.TITLE).style(self.STYLE)
+
+    def _headline(self, doc: Document):
+        doc.h1(self.TITLE)
+
+    def _file_table(self, doc: Document):
+        with doc.table(class_="autoindex"):
+            self._parent(doc)
+            for f in self._iter_files():
+                del f["priority"]
+                self._file_cell(doc, **f)
+
+    def _parent(self, doc: Document):
+        self._file_cell(doc, "📁", "..", "", "")
+
+    def _file_cell(
+        self,
+        doc: Document,
+        icon: str,
+        file_name: str,
+        file_access: str,
+        file_size: str,
+    ):
+        first = E.span(icon, class_="icon").a(file_name, href=file_name)
+        doc.tr.td(first).td(file_access).td(file_size)
+
+    def _prepare_file(self, path: Path) -> Dict[str, Union[int, str]]:
+        stat = path.stat()
+        modified = datetime.fromtimestamp(stat.st_mtime)
+        is_dir = S_ISDIR(stat.st_mode)
```

Comment:
pathlib.Path has `path.is_dir()`

Reply:
I figured since we already are grabbing the stat flags 🤷‍♂️
I did not benchmark the difference though, but it has to be at least something more. I'm pretty sure pathlib would do the same lookup.

Reply:
![image](https://user-images.githubusercontent.com/166269/214873330-2fd4cacb-acbd-4c95-a14b-11d18778824c.png)


Yup, under the hood, it is doing the same thing. Although, we may need to look at OS compat.